UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs

```diff
@@ -869,4 +871,140 @@ public void CmdRequestPaperEdit(GameObject paper, string newMsg)
 		}
 	}
 
+	/// <summary>
+	/// Performs a hug from one player to another.
+	/// </summary>
+	[Command]
+	public void CmdRequestHug(string hugger, GameObject huggedPlayer)
+	{
+		string huggee = huggedPlayer.GetComponent<PlayerScript>().playerName;
+		var huggedPlayerRegister = huggedPlayer.GetComponent<RegisterPlayer>();
+		ChatRelay.Instance.AddToChatLogServer(new ChatEvent
+		{
+			channels = ChatChannel.Local,
+			message = $"{hugger} has hugged {huggee}.",
+			position = huggedPlayerRegister.WorldPosition.To2Int()
+		});
+	}
+
+	/// <summary>
+	///	Performs a CPR action from one player to another.
+	/// </summary>
+	[Command]
+	public void CmdRequestCPR(GameObject rescuer, GameObject cardiacArrestPlayer)
+	{
+		var cardiacArrestPlayerRegister = cardiacArrestPlayer.GetComponent<RegisterPlayer>();
+
+		if (doingCPR)
+			return;
+
+		var progressFinishAction = new FinishProgressAction(
+			reason =>
+			{
+				switch (reason)
+				{
+					case FinishProgressAction.FinishReason.INTERRUPTED:
+						CancelCPR();
+						doingCPR = false;
+						break;
+					case FinishProgressAction.FinishReason.COMPLETED:
+						DoCPR(rescuer, cardiacArrestPlayer);
+						doingCPR = false;
+						break;
+				}
+			}
+		);
+
+		doingCPR = true;
+		UIManager.ProgressBar.StartProgress(cardiacArrestPlayerRegister.WorldPosition, 5f, progressFinishAction,
+			rescuer);
+		ChatRelay.Instance.AddToChatLogServer(new ChatEvent
+		{
+			channels = ChatChannel.Local,
+			message = $"{rescuer.Player()?.Name} is trying to perform CPR on {cardiacArrestPlayer.Player()?.Name}.",
+			position = cardiacArrestPlayerRegister.WorldPosition.To2Int()
+		});
+	}
+
+	[Server]
+	private void DoCPR(GameObject rescuer, GameObject CardiacArrestPlayer)
+	{
+		var CardiacArrestPlayerRegister = CardiacArrestPlayer.GetComponent<RegisterPlayer>();
+		CardiacArrestPlayer.GetComponent<PlayerHealth>().bloodSystem.oxygenDamage -= 7f;
+		doingCPR = false;
+		ChatRelay.Instance.AddToChatLogServer(new ChatEvent
+		{
+			channels = ChatChannel.Local,
+			message = $"{rescuer.Player()?.Name} has performed CPR on {CardiacArrestPlayer.Player()?.Name}.",
+			position = CardiacArrestPlayerRegister.WorldPositionServer.To2Int()
+		});
+	}
+
+	[Server]
+	private void CancelCPR()
+	{
+		// Stop the in progress CPR.
+		doingCPR = false;
+	}
+
+	/// <summary>
+	/// Performs a disarm attempt from one player to another.
+	/// </summary>
+	[Command]
+	public void CmdRequestDisarm(GameObject disarmer, GameObject playerToDisarm)
```

Comment:
apparently you can disarm other people while being dead, lol
also for some reason disarmed players keep their inhand sprites (at least when tested in multiplayer) so I'm not sure if `InventoryManager.DropItemInSlot()` is correct

Reply:
Ah turns out the function that drops an item in a specific slot is PlayerNetworkActions.DropItem(slot).
Fixed this in next commit.